### DISCUSSION TITLE
unacked tcp frames (tcp_window_full, unacked_len and send_data_total)

### DISCUSSION THREAD

**Discussion by lmikl** [zephyrproject-rtos/zephyr#67738] [Q&A] [answered]
I am currently working with a GD32F427 and an ADIN2111(ethernet/SPE) and I regularly have a problem with tcp_window_full.

On the GD32, I put zephyr with MQTT in QOS0. This card is connected via Ethernet to another device where there is a Mosquitto broker and an MQTT client.

When the Ethernet network is heavily loaded, I have frames that are not acknowledged (TCP ACK) by the card with the broker (verified via wireshark). After a while the communication blocks and the socket closes.
Following a debug, it seems that the tcp stack is unable to retransmit unacknowledged data and as a result the tcp_window_full function indicates that it is no longer possible to transmit.
If the Ethernet network is less loaded, i don't have the problem. 

If anyone knows which config word applied in order to forget the unacknowledged frames or if has anyone ever had this problem and resolved it?

**Comment by jukkar** [comment, marked as answer]
See https://docs.zephyrproject.org/latest/connectivity/networking/net_config_guide.html#tcp-options about setting the TX window size.
You should make the TX window smaller so that there is enough network buffers to send data and not just queue it. You could also increase the size of the TX net_buf's if there is enough RAM in the system.
